build_stats.sh

```diff
@@ -0,0 +1,40 @@
+#! /bin/sh
+
+set -e
+
+HERE_DIR="$(cd "$(dirname "$1")"; pwd)/$(basename "$1")"
```

Comment:
You're right, this is tricky. I knew a good trick for this at one point, but I've forgotten it :/ I know that we can't use `readlink -f` (which is the obvious way) because OS X doesn't have support.